core/src/main/java/hudson/model/Queue.java

```diff
@@ -1460,6 +1487,9 @@ public void maintain() {
             // allocate buildable jobs to executors
             for (BuildableItem p : new ArrayList<BuildableItem>(
                     buildables)) {// copy as we'll mutate the list in the loop
+                if (p.task instanceof FlyweightTask){
```

Comment:
@oleg-nenashev we just need the flyweighttask to be buildable so that the executor it is restricted to, can be provisioned, if offline. Once the executor is online, the makeBuildable function does the rest, as it was in the original implementation and the flyweighttask is removed from the buildable list. Originally the flyweighttask was not part of the buildable list whatsoever and the makebuildable function was responsible of executing it.